feature-libs/product-configurator/rulebased/components/attribute/types/input-field/configurator-attribute-input-field.component.ts

```diff
@@ -142,9 +171,16 @@ export class ConfiguratorAttributeInputFieldComponent
   }
 
   protected calculateDebounceTime(): number {
-    return (
-      this.config.productConfigurator?.updateDebounceTime?.input ??
-      this.FALLBACK_DEBOUNCE_TIME
-    );
+    if (this.isDateBased(this.attribute)) {
+      return this.debounceForDateActive
+        ? this.config.productConfigurator?.updateDebounceTime?.date ??
+            this.FALLBACK_DEBOUNCE_TIME_DATE
+        : 0;
+    } else {
+      return (
+        this.config.productConfigurator?.updateDebounceTime?.input ??
+        this.FALLBACK_DEBOUNCE_TIME
+      );
+    }
```

Comment:
Can we try to simplify this statement, maybe like this.
```suggestion
    return !this.isDateBased(this.attribute) || this.debounceForDateActive
      ? this.config.productConfigurator?.updateDebounceTime?.input ??
          this.FALLBACK_DEBOUNCE_TIME
      : 0;
```
because basically we say if it is date and and debounce is not active return 0, in all other case retuurn configured debounce time or fall back.

Reply:
No, that doesn't work, because we have different debounce times defined for input and date (as it's a bit more complex to enter a date value)